src/platform_impl/macos/window.rs

```diff
@@ -357,6 +358,29 @@ impl WinitWindow {
                     }
                 }
 
+                match attrs.parent_window {
+                    Some(RawWindowHandle::AppKit(handle)) => {
+                        // SAFETY: Caller ensures the pointer is valid or NULL
+                        let parent: Id<NSWindow, Shared> =
+                            unsafe { Id::retain(handle.ns_window.cast()) }.unwrap_or_else(|| {
+                                // SAFETY: Caller ensures the pointer is valid or NULL
+                                let parent_view: Id<NSView, Shared> =
+                                    unsafe { Id::retain(handle.ns_view.cast()) }
+                                        .expect("raw window handle should be non-empty");
+                                parent_view
+                                    .window()
+                                    .expect("parent view should be installed in a window")
```

Comment:
Could you propagate errors instead of doing `except`? You have `os_error` and such for that.